src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -435,8 +434,14 @@ public static String getReleaseVersionString()
         {
             JVMStabilityInspector.inspectThrowable(e);
             logger.warn("Unable to load version.properties", e);
-            return "debug version";
```

Comment:
This is a change of behavior - before, we were returning "`debug version`" here, now we will return null which leads to `UNKNOWN_RELEASE_VERSION`.
Also, this is not a bug fix but an improvement, so we should not commit it to 4.1, but only trunk.

Reply:
"debug_version" is nowhere referred. Its history dates back to 2010 and Cassandra 0.7!!! Non-null value was introduced to resolve some problems in IDE (https://issues.apache.org/jira/browse/CASSANDRA-1272). There is no way that we would run Cassandra without version.properties or in case it is unreadable. Also moving this method may be the actual fix because it eliminates to code path which led to the problem.